strawberryfields/backends/base.py

```diff
@@ -677,3 +680,34 @@ def state(self, modes=None, **kwargs):
             BaseGaussianState: state description
         """
         raise NotImplementedError
+
+
+# ==============================
+# Base bosonic backend
+# ==============================
+
+
+class BaseBosonic(BaseGaussian):
+    """Abstract base class for backends that are only capable of bosonic state manipulation."""
+
+    compiler = "bosonic"
```

Comment:
Unless explicitly needed at the moment, let's add this in when adding the compiler too so that it won't be forgotten that this is specified here.
```suggestion
```